Source/KSCrash/Recording/Tools/KSLogger.h

```diff
@@ -260,6 +260,14 @@ bool kslog_setLogFilename(const char* filename, bool overwrite);
 /** Clear the log file. */
 bool kslog_clearLogFile();
 
+/** Set whether or not to log to stdout.
+ *
+ * @param enabled Whether or not stdout logging should be enabled.
+ *
+ * By default stdout logging is enabled.
+ */
+void kslog_setLogToStdout(bool enabled);
```

Comment:
hmm that definitely shouldn't have happened. KSLogger was originally a standalone logging system that I used in a few projects, but eventually settled permanently in KSCrash, since it's really the only thing that needs such a slow level logging mechanism. So at first it had a "default" file descriptor (stdout) that it wrote to, and then if you called `kslog_setLogFilename`, it would replace that descriptor (so that if you never called it, you'd still get logging to stdout).

At one point I needed to check a problem where file descriptors weren't getting created on iOS but the file handle was still being considered "valid" by the OS, and so I did parallel logging to stdout and to a file. That code got checked in, which is why we're here now.

The original intent was either-or. stdout by default, or to a file of your choice; never both. The original mechanism allowed you to switch back and forth between stdout and a file by setting the field to null or non-null. I'm not sure if logging to both is ever going to be useful other than diagnosing an esoteric OS bug like I was doing in iOS 7.

So, to fix this, I think the best course of action would be:

- KSLogger by default points to stdout.
- No other part of KSCrash may set the file pointer unless it's passing along an explicit user request.
- If the user requests logging to a file, it writes to a file instead of stdout.
- If the user sets logfile to NULL, it logs to stdout.
- The log hooks are fine, but your choice of action will be basically limited to changing how you format the log messages, since to remain async-safe you can't do anything beyond writing to files using the low level `write()`, exactly like KSLogger is doing now.